src/code/z_sram.c

```diff
@@ -1,25 +1,884 @@
+typedef struct {
+    /* 0x0000 */ SavePlayerData playerData; // "S_Private" substruct name
+    /* 0x004C */ ItemEquips equips;
+    /* 0x0058 */ Inventory inventory;
+    /* 0x00B8 */ SavedSceneFlags sceneFlags[124];
+    /* 0x0E48 */ FaroresWindData fw;
+    /* 0x0E70 */ char unk_E70[0x10];
+    /* 0x0E80 */ s32 gsFlags[6];
+    /* 0x0E98 */ char unk_E98[0x10];
+    /* 0x0EA8 */ s32 horseRaceRecord;
+    /* 0x0EAC */ char unk_EAC[0x0C];
+    /* 0x0EB8 */ u16 eventChkInf[14]; // "event_chk_inf"
+    /* 0x0ED4 */ u16 itemGetInf[4];   // "item_get_inf"
+    /* 0x0EDC */ u16 infTable[30];    // "inf_table"
+    /* 0x0F18 */ char unk_F18[0x04];
+    /* 0x0F1C */ u32 worldMapAreaData; // "area_arrival"
+    /* 0x0F20 */ char unk_F20[0x4];
+    /* 0x0F24 */ u8 scarecrowCustomSongSet;
+    /* 0x0F25 */ u8 scarecrowCustomSong[0x360];
+    /* 0x1285 */ char unk_1285[0x24];
+    /* 0x12A9 */ u8 scarecrowSpawnSongSet;
+    /* 0x12AA */ u8 scarecrowSpawnSong[0x80];
+    /* 0x132A */ char unk_132A[0x02];
+    /* 0x132C */ HorseData horseData;
+    /* 0x1336 */ u16 checksum; // "check_sum"
+} SaveInfo;                    // size = 0x1338
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram/func_800A88D4.s")
+typedef struct {
+    /* 0x00 */ s32 entranceIndex;
+    /* 0x04 */ s32 linkAge; // 0: Adult; 1: Child
+    /* 0x08 */ s32 cutsceneIndex;
+    /* 0x0C */ u16 dayTime; // "zelda_time"
+    /* 0x10 */ s32 nightFlag;
+    /* 0x14 */ s32 numDays;
+    /* 0x18 */ s32 unk_18;    // increments with numDays, gets reset by goron for bgs and one other use
+    /* 0x1C */ SaveInfo info; // "information"
+} Save;                       // size = 0x1354
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram/func_800A8A20.s")
+#define SAVE_PLAYER_DATA (*((SavePlayerData*)&gSaveContext.newf))
+#define SAVE_INFO (*((SaveInfo*)&gSaveContext.newf))
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram/func_800A9258.s")
+#define FIELD_SIZE(type, field) sizeof(((type*)0)->field)
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram/func_800A96D0.s")
+#define SRAM_SIZE 0x8000
+#define SLOT_SIZE 0x1450
+#define CHECKSUM_SIZE 0x9AAu
+#define SAVE_SIZE 0x1354 // FIELD_SIZE(SaveContext, save)
+#define INFO_SIZE 0x1338 // FIELD_SIZE(SaveContext, save.info)
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram/func_800A97F0.s")
+#define DEATHS OFFSETOF(SaveContext, deaths)
+#define NAME OFFSETOF(SaveContext, playerName)
+#define N64DD OFFSETOF(SaveContext, n64ddFlag)
+#define HEALTH_CAP OFFSETOF(SaveContext, healthCapacity)
+#define QUEST OFFSETOF(SaveContext, inventory.questItems)
+#define DEFENSE OFFSETOF(SaveContext, inventory.defenseHearts)
+#define HEALTH OFFSETOF(SaveContext, health)
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram/func_800A9A9C.s")
+#define SLOT_OFFSET(idx) (SLOT_SIZE * idx + 0x20)
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram/func_800A9AD0.s")
+#define CALC_CHECKSUM(i, j, ptr, sum)         \
```

Comment:
Really not a fan of putting code inside macros like that, unless it's necessary ofc, but maybe that's just me.

Reply:
@Random06457 thoughts, i dont really mind either way. if file select is going to be doing a checksum calc the same way the macro might be useful to have, but idk if it will

Reply:
Also I could be wrong but it looks like there are other slightly different loops to calculate checksums in the file that don't fall under this same macro so that's another reason to not have it imo, since it can't be consistent.